### PR TITLE
refactor pushqueue

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

Refactor PushQueue

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -665,8 +665,9 @@ func (s *DiscoveryServer) startPush(push *model.PushContext, full bool, edsUpdat
 		adsLog.Infof("Starting new push while %v were still pending", currentlyPending)
 	}
 	startTime := time.Now()
+	pushEvent := &PushEvent{edsUpdates, push, startTime, full}
 	for _, p := range pending {
-		s.pushQueue.Enqueue(p, &PushInformation{edsUpdates, push, startTime, full})
+		s.pushQueue.Enqueue(p, pushEvent)
```
Comment (howardjohn) [inline pilot/pkg/proxy/envoy/v2/ads.go]: This is broken, each event push event needs to be distinct or you will have race conditions. Please keep it in the loop

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/ads.go]: Let me have a look

Comment (hzxuzhonghu) [reply pilot/pkg/proxy/envoy/v2/ads.go]: I think i know what you mean, it may happen when the queue accumulated

**Pull request merged**